addons/account/models/chart_template.py

```diff
@@ -279,6 +279,13 @@ def _load(self, sale_tax_rate, purchase_tax_rate, company):
         # If the floats for sale/purchase rates have been filled, create templates from them
         self._create_tax_templates_from_rates(company.id, sale_tax_rate, purchase_tax_rate)
 
+        # Set the fiscal country before generating taxes in case the company does not have a country_id set yet
+        if company.country_id and not self.country_id:
+            # Otherwise, if there is a country set on the company, set the default fiscal country of the company equal to the company country.
+            company.account_fiscal_country_id = company.country_id
+        else:
+            company.account_fiscal_country_id = self.env.ref('base.us')
```

Comment:
This will happen when not company.country_id or self.country_id ... Seems wrong.

To me, this should be run if not company.country_id and not self.country_id, hence making sure that a country will always be chosen. 

You probably want to move the condition from line 347 to tthis region of the code as well. They have the same job, after all: setting the fiscal country.

So, we'd have 

if self.country_id:
# set fiscal country to self.country_id
elif not self.account_fiscal_country_id;
# set US by default

Then, you need to install all taxes for the fiscal country set on the company (which might require a few changes in subfunctions).